### PR TITLE
fix duplicate line_length lints.

### LINKED ISSUE #681: Duplicate lints for `line_length_linter`
Found during vignette writing for #660
``` r
lintr::lint("x <- 2 # -------------------------------------------------------------------------------\n", 
            parse_settings = FALSE)
#> <text>:1:1: style: Lines should not be more than 80 characters.
#> x <- 2 # -------------------------------------------------------------------------------
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#> <text>:1:1: style: Lines should not be more than 80 characters.
#> x <- 2 # -------------------------------------------------------------------------------
#> ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

<sup>Created on 2020-12-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Happens only if the offending line contains R code.
I assume the problem is we lint over `full_parsed_content` plus `parsed_content`.

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #681

also added a regression test.

### CODE REVIEW

File: R/line_length_linter.R [hunk 1]
```diff
@@ -4,10 +4,14 @@
 line_length_linter <- function(length) {
   function(source_file) {
 
-    lapply(names(source_file$lines)[vapply(source_file$lines, nchar, integer(1)) > length],
+    # Only go over complete file
+    if (is.null(source_file$file_lines)) return(list())
+
+    lapply(which(vapply(source_file$file_lines, nchar, integer(1)) > length),
```
Comment (russHyde) [inline R/line_length_linter.R:10]: Could you split this statement for clarity `oversized_lines <- which(vapply(...)); lapply(oversized_lines, function(line_number) {...})`

File: tests/testthat/test-line_length_linter.R [hunk 2]
```diff
@@ -30,4 +29,11 @@ test_that("returns the correct linting", {
   expect_lint("aaaaaaaaaaaaaaaaaaaab",
     rex("Lines should not be more than 20 characters"),
     line_length_linter(20))
+
+  # Don't duplicate lints
+  expect_length(
+    lint("x <- 2 # ------------\n",
```
Comment (russHyde) [inline tests/testthat/test-line_length_linter.R:35]: In tests, calls to 'lint' should have `parse_settings = FALSE` shouldn't they?

**Review by russHyde — CHANGES_REQUESTED** [review]

File: tests/testthat/test-line_length_linter.R (hunk 2 above)
Comment (AshesITR) [reply tests/testthat/test-line_length_linter.R:35]: absolutely!

**Review by russHyde — APPROVED** [review]
LGTM

**Pull request merged**